src/components/Kanban.tsx

```diff
@@ -0,0 +1,311 @@
+  const cards = React.useMemo(() => {
+    const cards: Record<string, Result[]> = {};
+    data.forEach((d) => {
+      const column = d[columnKey]?.toString() || `No ${columnKey}`;
+      if (!cards[column]) {
+        cards[column] = [];
+      }
+      cards[column].push(d);
+    });
+    Object.keys(cards).forEach((k) => {
+      cards[k] = cards[k].sort(
+        (a, b) => prioritization[a.uid] - prioritization[b.uid]
+      );
+    });
+    return cards;
+  }, [data, prioritization, columnKey]);
+  const potentialColumns = React.useMemo(() => {
+    const columnSet = new Set(columns);
+    return Object.keys(cards).filter((c) => !columnSet.has(c));
+  }, [cards, columns]);
+  React.useEffect(() => {
+    const base64 = window.btoa(JSON.stringify(prioritization));
+    setInputSetting({
+      blockUid: layoutUid,
+      key: "prioritization",
+      value: base64,
+    });
+  }, [prioritization]);
+  const containerRef = React.useRef<HTMLDivElement>(null);
+  const reprioritize = React.useCallback<Reprioritize>(
+    ({ uid, x, y }) => {
+      if (!containerRef.current) return;
+      // use x & y to determine priority
+      const newColumn = Array.from(
+        containerRef.current.querySelectorAll(".roamjs-kanban-column")
+      )
+        .reverse()
+        .find((el) => {
+          const { left } = el.getBoundingClientRect();
+          return x >= left;
+        });
+      if (!newColumn) return;
+
+      const column = newColumn.getAttribute("data-column");
+      if (!column) return;
+
+      const _cardIndex = Array.from(
+        newColumn.querySelectorAll(
+          ".roamjs-kanban-card:not(.react-draggable-dragging)"
+        )
+      )
+        .map((el, index) => ({ el, index }))
+        .reverse()
+        .find(({ el }) => {
+          const { top } = el.getBoundingClientRect();
+          return y >= top;
+        })?.index;
+      const cardIndex = typeof _cardIndex === "undefined" ? -1 : _cardIndex;
+      const topCard = cards[column]?.[cardIndex];
+      const bottomCard = cards[column]?.[cardIndex + 1];
+      const topPriority = prioritization[topCard?.uid] || 0;
+      const bottomPriority = prioritization[bottomCard?.uid] || 1;
+      const priority = (topPriority + bottomPriority) / 2;
+      setPrioritization((p) => ({ ...p, [uid]: priority }));
+
+      const result = byUid[uid];
+      if (!result) return;
+      const columnKeySelection = resultKeys.find(
+        (rk) => rk.key === columnKey
+      )?.selection;
+      if (!columnKeySelection) return;
+      const predefinedSelection = predefinedSelections.find((ps) =>
+        ps.test.test(columnKeySelection)
+      );
+      if (!predefinedSelection?.update) return;
+      const { [`${columnKey}-uid`]: columnUid } = result;
+      if (typeof columnUid !== "string") return;
+      predefinedSelection
+        .update({
+          uid: columnUid,
+          value: column,
+          selection: columnKeySelection,
+        })
+        .then(onQuery);
```

Comment:
This is going to be the hardest part of this feature - updating our Roam Graph deterministically based on how the user queried the data